### DISCUSSION TITLE
Explain difference in `dag_longest_path()` output between v2.5 and v2.6+

### DISCUSSION THREAD

**Discussion by veghp** [networkx/networkx#5712] [Q&A] [answered]
Thanks for this great package that we use in our GeneBlocks package.

I have a question re different outputs for dag_longest_path():

Running the below on networkx v2.5:
```python
import networkx as nx

test_graph = nx.DiGraph([('block_1', 'block_3'), ('block_1', 'block_2')])
nx.dag_longest_path(test_graph)
# Returns:
# ['block_1', 'block_2']
```

Running on networkx v2.8:
```python
import networkx as nx
test_graph = nx.DiGraph([('block_1', 'block_3'), ('block_1', 'block_2')])
nx.dag_longest_path(test_graph)
# Returns:
# ['block_1', 'block_3']
```

Is there an explanation for the difference? A look at the changelist doesn't explain the change (the change/difference appears at v2.6): https://networkx.org/documentation/stable/release/release_2.6.html

Both ran on Python v3.9 (but have the same result on Python v3.6 / networkx 2.5).

**Comment by dschult** [comment, marked as answer]
The difference is due to #4757 which has a side effect of reversing the order of the topological sort used in `dag_longest_path`. Both orders are topological sorts, and indeed, both results displayed in your example are longest paths.

**Reply by rossbar** [reply]
Note that there is a `topo_order` kwarg to `dag_longest_path` that allows you to pass in a *preferred* topological ordering in the case where there are multiple.

One possible improvement that comes to mind would be to add a note and/or example to the docstring for the case where there are multiple valid DAG longest paths.

**Reply by dschult** [reply]
That is a good suggestion for a new example showing the `topo_order` kwarg.

**Reply by veghp** [reply]
Thank you for the quick answer and explanation.

**Reply by dschult** [reply]
#5728 added an example of using the `topo_order` kwarg.
